packages/lib/src/user/utils.ts

```diff
@@ -252,30 +258,32 @@ const hasAudience = (
   // The user affiliation was found but not the campus, don't show this
   if (!hasValue(locations, campusName)) return false;
 
+  // The user affiliation and campus were found, audiences empty is treated as reason to show all and should
+  // skip any further evaluation.
+  if (audiences?.length === 0) return true;
```

Comment:
The bug we encountered was because this return wasn't being considered before the (now removed) `item.audiences.some..` on line 264 in the old code.